app/src/main/java/org/openobservatory/ooniprobe/activity/PromptActivity.java

```diff
@@ -0,0 +1,209 @@
+
+import javax.inject.Inject;
+
+public class PromptActivity extends AbstractActivity {
+    private static final String PROMPT_ITEM = "promptItem";
+    @Inject
+    UpdatesNotificationManager notificationManager;
+    private ActivityPromptBinding binding;
+    private Prompt prompt;
+
+    private ActivityResultLauncher<String> requestPermissionLauncher;
+
+    public static Intent newIntent(Context context, Prompt prompt) {
+        return new Intent(context, PromptActivity.class).putExtra(PROMPT_ITEM, prompt);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        getActivityComponent().inject(this);
+        binding = ActivityPromptBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        prompt = (Prompt) getIntent().getExtras().get(PROMPT_ITEM);
+        binding.title.setText(prompt.title);
+        binding.description.setText(prompt.paragraph);
+
+        registerPermissionRequest();
+        setUpClickListeners();
+    }
+
+    private void registerPermissionRequest() {
+        requestPermissionLauncher = registerForActivityResult(new ActivityResultContracts.RequestPermission(), (result) -> {
+            if (!result) {
+                Snackbar.make(binding.getRoot(), "Please grant Notification permission from App Settings", Snackbar.LENGTH_LONG).setAction(R.string.Settings_Title, view -> {
+                    Intent intent = new Intent();
+                    intent.setAction("android.settings.APP_NOTIFICATION_SETTINGS");
+                    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+                    //for Android 5-7
+                    intent.putExtra("app_package", getPackageName());
+                    intent.putExtra("app_uid", getApplicationInfo().uid);
+
+                    // for Android 8 and above
+                    intent.putExtra("android.provider.extra.APP_PACKAGE", getPackageName());
+
+                    startActivity(intent);
+                }).show();
+            }
+            setResult(result ? Activity.RESULT_OK : Activity.RESULT_CANCELED);
+            finish();
+        });
+    }
+
+    private void setUpClickListeners() {
+        OnPromptAction actions;
+        switch (prompt) {
+            case CENSORSHIP_CONSENT:
+                actions = new InternetCensorshipConsentActions();
+                break;
+            case TEST_PROGRESS_CONSENT:
+                actions = new TestProgressNotificationConsentActions();
+                break;
+            default:
+                actions = new OnPromptAction() {
+                    @Override
+                    public void onClickPositive(View view) {
+                    }
+
+                    @Override
+                    public void onClickNeutral(View view) {
+                    }
+
+                    @Override
+                    public void onClickNegative(View view) {
+                    }
```

Comment:
If those methods are left intentionally blank, please leave a comment like `// nothing` to imply that.